src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    public void getStatistic(String fromFile, String toFile) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFile))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (parts.length != 2) {
+                    continue;
+                }
+                String operation = parts[0].trim();
+                int amount = Integer.parseInt(parts[1].trim());
+
+                if (operation.equals("supply")) {
+                    supplyTotal += amount;
+                } else if (operation.equals("buy")) {
+                    buyTotal += amount;
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Error reading file" + e.getMessage());
+        }
```

The catch block for reading prints an error instead of rethrowing a contextual RuntimeException. This violates the checklist: "Exceptions must not be ignored; do not leave empty catch blocks or only print stack traces. Re-throw a RuntimeException with a meaningful message including the file name and the original exception as the cause [CHECKLIST ITEM #12]." Replace the `System.out.println` with `throw new RuntimeException("Can't read data from file " + fromFile, e);` to preserve the cause and provide context (file name). See the read catch block here .